### PR TITLE
Datepicker builder

### PR DESCRIPTION
**Pull request by isaacfink**
Added date picker builder, there is still some housekeeping to do but the structure is there, I am opening this now because I am taking a break and might not get back to it for a while

### CODE REVIEW

**Comment by isaacfink** [comment]
I still need to do the following

- [ ] aria labels
- [ ] respect earliest and latest date and disabled
- [ ] confirm choice (should not automatically change value on select)
- [ ] add range support
- [ ] write docs
- [ ] clean up and write tests (at least for utils)

File: package-lock.json
Comment (TGlide) [inline package-lock.json:1]: You should be usin pnpm, not npm! Please be sure to remove this file, then use pnpm install instead.

Comment (isaacfink) [reply package-lock.json:1]: @TGlide I will be sure to do that before it's completed, I was actually wondering how come there is no lockfile

**Comment by ollema** [comment]
this should be closed now I think?

**Pull request closed without merging**